src/duplicacy_backupmanager.go

```diff
@@ -288,6 +288,31 @@ func (manager *BackupManager) Backup(top string, quickMode bool, threads int, ta
 	var modifiedEntries []*Entry  // Files that has been modified or newly created
 	var preservedEntries []*Entry // Files unchanges
 
+	// identify small files based on their hash to avoid bundled uploads during renames
+	fileHashes := make(map[string]*Entry)
+	fileBuffer := make([]byte, 64*1024)
+	minSize := int64(manager.config.MinimumChunkSize)
+
+	checkHash := func(local *Entry) {
+		if local.Size < minSize && local.Hash == "" {
+			// compute hash for small file
+			local.Hash = manager.config.ComputeFileHash(local.Path, fileBuffer)
+		}
+		remote := fileHashes[local.Hash]
+		if remote != nil {
+			// hash of file matches a previous file
+			LOG_DEBUG("FILE_HASH", "Hash of file %s matches previously known file %s", local.Path, remote.Path)
+			local.StartChunk = remote.StartChunk
+			local.StartOffset = remote.StartOffset
+			local.EndChunk = remote.EndChunk
+			local.EndOffset = remote.EndOffset
+			preservedEntries = append(preservedEntries, local)
```

Comment:
The chunks are added to `preservedChunkHashes` from the `preservedEntries` a little bit further down (line 383 onward), so that part is already taken care of. 

Files smaller than MinimumChunkSize are never split over several chunks. Processing them in two passes and using file boundaries to form chunks ensures that. You could probably add an assert to really guard against someone breaking that in the future..